lib/grovepi/poller.ex

```diff
@@ -1,4 +1,53 @@
 defmodule GrovePi.Poller do
+  @moduledoc """
+  A behaviour module for implementing polling on a pin.
+
+  The GrovePi.Poller behavior abstracts polling on a pin.  Developers are
+  responsible for implementing `c:read_value/2` and a module using the
+  `GrovePi.Trigger` behaviour.
+
+  ## Example
+
+  This example shows implementation of `GrovePi.Potentiometer`.  The module
+  should `use GrovePi.Poller`, specifiying the `:default_trigger` and
+  `:read_type`.  It should have a `c:read_value/2` callback which reads the
+  desired sensor.
+
+      defmodule GrovePi.Potentiometer do
+        alias GrovePi.Analog
+
+        use GrovePi.Poller, default_trigger: GrovePi.Potentiometer.DefaultTrigger,
+          read_type: Analog.adc_level
+
+        def read_value(prefix, pin) do
+          Analog.read(prefix, pin)
+        end
+      end
+
+  The following is implementation of `GrovePi.Potentiometer.DefaultTrigger`. The
```

Comment:
Instead of the trigger example here I think we should just link to the trigger docs. That way if we change something about the trigger we only need to change docs in one place.